tests/unit/test__fixture__cmdln_args.py

```diff
@@ -0,0 +1,161 @@
+from unittest.mock import patch
+
+from tests.unit.conftest import (
+    CONFTEST_TEMPLATE,
+    PYFILE_ASSERT_EMPTY_CONFIG,
+    PYFILE_ASSERT_EXPECTED_CONFIG,
+    PYFILE_TEMPLATE,
+    perform_fixture_test_with_optional_log_capture,
+)
+
+from grasshopper.lib.configuration.gh_configuration import (  # noqa: N817
+    ConfigurationConstants as CC,
+)
+
+FIXTURE_UNDER_TEST = "cmdln_args"
+
+# Some comments on testing fixture cmdln_args.
+# The source of values for this fixture is pytest's request.config object. We are going
+# on the assumption that this fixture is well tested by the pytest package and are only
+# checking what essentially are positive test cases for this collection.
+# Note, we are not using magic mock here because pytest's request.config object is very
+# hard to patch. See the definition of MockConfig (conftest) for more details.
+
+
+def test__cmdln_args__happy(pytester):
+    """Fixture should load any matching values from the command line."""
+
+    patches = """
+        from tests.unit.conftest import MockConfig
+
+        @pytest.fixture(scope="session")
+        def request_config():
+            mock = MockConfig({"--users": 42, "--spawn_rate":.3})
+            return mock
+        """
+    pytester.makeconftest(
```

Comment:
This is a really cool pattern, is this something you discovered while at pycon? I am interested to learn more about it.

Reply:
I had learned just before pycon about how to use pytester along with makepyfile and makeconftest in order to test a plugin (which for testing purposes, is anything that includes pytest fixtures. this is because of how pytest loads the fixtures before your test can intervene and patch or mock anything).

However, it was on the train home that I thought up the way of using template strings to make the boilerplate a lot smaller while still keeping the important parts - what am I patching, what am I validating - visible here in the test case. 

You can also load conftest and pyfiles from files saved on disk, but I didn't want to 'hide' what the test was doing, so  I went this route. I am planning on giving 2 lunch and learns coming up about unit testing. The first unit testing design and basics, the second on advanced features like mock/patch/pytester. But in the mean time, we can chat about this if you want.